### PR TITLE
fixes discogs track-parsing for only one artist

### PR DESCRIPTION
**Pull request by funkyfuture**
this fixes the case that fetched artist-fields from discogs would always have a ',' as tail.

i also did some occasional code cleanup. i hope you don't mind.

on that occasion: what do you think about porting puddletag to python 3?

### CODE REVIEW

File: source/puddlestuff/tagsources/discogs.py [hunk 1]
```diff
@@ -290,12 +297,13 @@ def search(artist=None, album=None):
         keywords = album
     return keyword_search(keywords)
 
+
 def urlopen(url):
     url = iri_to_uri(url)
     request = urllib2.Request(url)
     request.add_header('Accept-Encoding', 'gzip')
-    if puddlestuff.tagsources.user_agent:
```
Comment (keithgg) [inline source/puddlestuff/tagsources/discogs.py]: Leave this as, the user agent could get changed (bad form, I know, but I wrote the code 5 years ago). 

If you import the user agent only once, it won't get updated if the user updates their preferences.

**Comment by keithgg** [comment]
Left a comment. If you fix it, I'll merge this in.

**Comment by funkyfuture** [comment]
okay, i amended a function to get the currently set useragent. also stumbled over other issues on the way.

**Comment by keithgg** [comment]
Thanks!

**Pull request merged**